packages/sync-actions/src/product-actions.js

```diff
@@ -61,10 +61,15 @@ export function actionsMapVariants (diff, oldObj, newObj) {
       ...newObject,
       action: 'addVariant',
     }),
-    [REMOVE_ACTIONS]: objectToRemove => ({
-      action: 'removeVariant',
-      id: objectToRemove.id,
-    }),
+    [REMOVE_ACTIONS]: ({ id }, index) => {
+      // The master variant can not be removed
```

Comment:
We can't remove the master variant. This was done before by treating it as in the `masterVariant` property which would non run through the array detection process.